src/dirichlet.cpp

```diff
@@ -57,7 +74,18 @@ void Linear_Function::eval (
   dolfin::Array<double>& values,
   const dolfin::Array<double>& x
 ) const {
-    values[0] = _lower_value * (_mesh_max - x[_coordinate]) * _distance;
-    values[0] += _upper_value * (x[_coordinate] - _mesh_min) * _distance;
+    if (_dimension==1)
```

Comment:
Why is this conditional needed? If `_dimension == 1`, then the forloop in the `else` statement will only run for `i = 0`.